### ISSUE TITLE
Missing Class Docstring false positive when inheriting from a generic class of a TypedDict instance

### ISSUE THREAD

**Issue by TNTzx** [PyCQA/pylint#7934]
### Bug description

```py
"""Module Doc"""

import typing as typ
import abc

T_Test = typ.TypeVar("T_Test")
class Bases(typ.Generic[T_Test]):
    """Class Doc"""
    @abc.abstractmethod
    def method(self) -> T_Test:
        """Method Doc"""


typed_dict = typ.TypedDict("typed_dict", {"test": int})
class Child(Bases[typed_dict]):
    """Class Doc"""
    def method(self): # missing-function-docstring
        return {"test": 0}
```

Note that the false positive doesn't happen when you use the class form of `TypedDict`:
```py
class TypedDictExample(typ.TypedDict):
    """Another Class Doc"""
    test: int

class Child(Bases[TypedDictExample]):
    """Class Doc"""
    def method(self): # no error!
        return {"test": 0}
```

### Command used

```shell
python.exe -m pylint --reports=n --output-format=json --variable-rgx=[a-z_][a-z0-9_]{0,30}$ --class-rgx=[A-Z_]?(([a-zA-Z0-9]|_)+)$ --function-rgx=[a-z_][a-z0-9_]{0,30}$ --attr-rgx=[a-z_][a-z0-9_]{0,30}$ --argument-rgx=[a-z_][a-z0-9_]{0,30}$ --class-attribute-rgx=[a-z_][a-z0-9_]{0,30}$ --disable=line-too-long,super-init-not-called,unused-import,consider-using-from-import,unused-argument,too-few-public-methods,too-many-public-methods,too-many-ancestors,too-many-arguments,too-many-instance-attributes,too-many-statements,too-many-branches,too-many-return-statements,too-many-locals,assigning-non-slot,fixme --extension-pkg-allow-list=win32 --typevar-rgx=[A-Z]|(T_([A-Za-z]+))$ --from-stdin main.py
```

### Pylint output

```shell
For the first Code block:

[
    {
        "type": "convention",
        "module": "main",
        "obj": "Child.method",
        "line": 17,
        "column": 4,
        "endLine": 141,
        "endColumn": 14,
        "path": "testcode.py",
        "symbol": "missing-function-docstring",
        "message": "Missing function or method docstring",
        "message-id": "C0116"
    }
]

For the second code block:
```json
[]
```
```


### Expected behavior

No error

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.10.1 (tags/v3.10.1:2cd268a, Dec  6 2021, 19:10:37) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 11
